src/platform/stm32f1/spi.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2021 Rafael Silva <silvagracarafael@gmail.com>
+ */
+
+#pragma once
+
+#include <stm32f1xx.h>
+
+#include "platform/stm32f1/gpio.h"
+
+#define SPI_MSB_FIRST 0
+#define SPI_LSB_FIRST 1
+
+#define SPI_MODE0 0
+#define SPI_MODE1 1
+#define SPI_MODE2 2
+#define SPI_MODE3 3
+
+struct spi_drv_data_t {
+	struct gpio_pin_t cs_gpio;
+	u8 cs_inverted;
+	SPI_TypeDef *interface;
+};
+
+s8 spi_init(
```

Comment:
Let's make this `int` instead of `s8`. Only use specific sizes when there is a specific need, here it we just want an int.

Reply:
ok